### PR TITLE
POLIO-1194 correct process for mail vaccine auth cron jobs

### PR DESCRIPTION
**Pull request by SimonDus**
The cronjobs were not linked to a task. This PR fix it.

Related JIRA tickets : POLIO-1194

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [x] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

The function are now a task launched by a cron.

## How to test

Quite hard to test on a local setup as it's directly linked Elastic Beanstalk
Check the code, the tests or run the function directly from a python shell.

### CODE REVIEW

File: plugins/polio/tasks/vaccine_authorizations_mail_alerts.py [hunk 1]
```diff
@@ -136,9 +147,54 @@ def expired_vaccine_authorizations_email_alert():
     return {"vacc_auth_mail_sent_to": mailing_list} if mail_sent else "no_vacc_auth_mail_sent"
 
 
-def vaccine_authorization_update_expired_entries():
+@task_decorator(task_name="expired_vaccine_authorizations_email_alert")
+def send_email_expired_vaccine_authorizations_alert(task=None):
+    past_date = dt.date.today() - timedelta(days=1)
+    vaccine_auths = VaccineAuthorization.objects.filter(expiration_date=past_date)
+    team = get_object_or_404(Team, name="nOPV2 vaccine authorization alerts")
```
Comment (kemar) [inline plugins/polio/tasks/vaccine_authorizations_mail_alerts.py]: Is it OK to hardcode `name` here?

If so you can move the `team` query before `vaccine_auths`, so you save one query if there is no `team`.

File: plugins/polio/tasks/vaccine_authorizations_mail_alerts.py [hunk 2]
```diff
@@ -136,9 +147,54 @@ def expired_vaccine_authorizations_email_alert():
...
+    mailing_list = [user.email for user in User.objects.filter(pk__in=team.users.all())]
```
Comment (kemar) [inline plugins/polio/tasks/vaccine_authorizations_mail_alerts.py]:
```suggestion
    mailing_list = team.users.values_list('pk', flat=True)
```

Another way to do the same thing.

File: plugins/polio/tasks/vaccine_authorizations_mail_alerts.py [hunk 3]
```diff
@@ -136,9 +147,54 @@ def expired_vaccine_authorizations_email_alert():
...
+        task.report_progress_and_stop_if_killed(
+            progress_value=i,
+            end_value=total,
+            progress_message=f"Campaign {vacc_auth.pk} started",
+        )
+
+        logger.info(f"Email for {vacc_auth}")
+        status = vaccine_authorizations_60_days_expiration_email_alert(vaccine_auths, mailing_list)
+        if not status:
+            logger.info(f"... skipped")
```
Comment (kemar) [inline plugins/polio/tasks/vaccine_authorizations_mail_alerts.py:171]: Does "skipped" mean that the sending failed? Shouldn't we log that at the `error` level?

File: plugins/polio/tests/test_vaccine_authorizations.py [hunk 4]
```diff
@@ -522,11 +546,14 @@ def test_expired_vaccine_authorizations_email_alert(self):
...
-        response = expired_vaccine_authorizations_email_alert()
+        # test the function itself to check if the content is correct
+
+        response = expired_vaccine_authorizations_email_alert(vaccine_auths, mailing_list)
 
-        page_url = f"example.com//dashboard/polio/vaccinemodule/nopv2authorisation/accountId/{team.project.account.id}/order/-current_expiration_date/pageSize/20/page/1"
+        page_url = f"example.com/dashboard/polio/vaccinemodule/nopv2authorisation/accountId/{self.team.project.account.id}/order/-current_expiration_date/pageSize/20/page/1"
         url_is_correct = False
 
+        print(mail.outbox[0].body)
```
Comment (kemar) [inline plugins/polio/tests/test_vaccine_authorizations.py]: `print()` statement to delete?

**Review by kemar — COMMENTED** [review]
Some basic feedback since I don't know anything about Polio yet.

File: plugins/polio/tasks/vaccine_authorizations_mail_alerts.py (hunk 1 above)
Comment (SimonDus) [reply plugins/polio/tasks/vaccine_authorizations_mail_alerts.py]: Yes, this is the only team that must receive the notifications.

**Review by kemar — APPROVED** [review]
LGTM AFAICT

**Pull request merged**